strax/chunk.py

```diff
@@ -65,7 +65,7 @@ def __init__(self,
 
         if len(self.data):
             data_starts_at = self.data[0]['time']
-            data_ends_at = strax.endtime(self.data[-1])
+            data_ends_at = strax.endtime(self.data).max()
```

Comment:
Can we apply here the same logic as for raw_records, but I am afraid not since peaks have no fixed length?

Reply:
For records we can make arguments about the number of channels because they overlap in time, but peaks don't. The second-last peak has to end before the last peak begins.

Reply:
For peaks this should really be peanuts to compute as there are so few (relative to raw_records)

Reply:
I am wondering if our improved approach in https://github.com/XENONnT/straxen/pull/146 becomes obsolete with this change. Was this change included in your timing benchmark?

Reply:
No I didn't include it in the benchmark. 

Further, the computation here is different from the one in https://github.com/XENONnT/straxen/pull/146. Here this number is only used to make the ValueError (such that we don't make mistakes like https://github.com/XENONnT/straxen/pull/146) but hit the ValueError if we do try.

Reply:
I agree, my concern was also more regarding the computation time. Either way I guess we do not have much other options.

Reply:
Sure, I guess it will never be worse than for raw_records but it'd be good to try.

Reply:
Maybe I am wrong but this check would be applied for all data_kinds hence you add again your benchmarked  84.7 ms/chunk for raw_record and record etc at least. Is not it?

Reply:
Okay, so this is the deal:
I did a run with 1.6+0.4 s chunking (ergo many checks).

**Before update**
3211346 MainThread root bootstrax (eb0.xenon.local) finished run 8600 in 3204 seconds

**After update**
3477742 MainThread root bootstrax (eb0.xenon.local) finished run 8600 in 3471 seconds

I think this is an unacceptable loss of performance for a check that shouldn't be hit for properly constructed chunks. I'd argue that for a little less safety we might want to limit looking back the last say _500_ samples? This would be an arbitrary number but we have been running so long with just 1 sample I would be surprised if we ever ever hit this problem at sample [:-501]. 

I'm running this test again to see how reproducible these results are and will edit these numbers into this comment.